### PR TITLE
[REF] web: pivot_view convert to hoot

### PR DESCRIPTION
**Pull request by pipu-odoo**
In this commit, qunit tests of pivot view are converted in hoot
unit tests.

task~3705027

### CODE REVIEW

**Review by Arcasias — CHANGES_REQUESTED** [review]

**Comment by pipu-odoo** [comment]
@Arcasias Thank you very much for you review and comments ! Bismi

**Review by Arcasias — COMMENTED** [review]
Sorry, missed one :weary:

File: addons/web/static/tests/views/pivot_view.test.js [hunk 1]
```diff
@@ -0,0 +1,4553 @@
...
+    await contains("thead th.o_pivot_header_cell_closed:eq(1)").click();
+
+    expect(".o-dropdown--menu").toHaveCount(1);
+});
+
+test("Server order is kept by default", async () => {
+    expect.assertions(1);
+
+    let isSecondReadGroup = false;
+    onRpc(({ method }) => {
```
[the remark this answers is not in the document]
Comment (pipu-odoo) [reply addons/web/static/tests/views/pivot_view.test.js:3543]: thx

File: addons/web/static/tests/views/pivot_view.test.js [hunk 2]
```diff
@@ -0,0 +1,4551 @@
...
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        context: { search_default_small_than_0: true },
+        noContentHelp: markup(`<p class="abc">click to add a foo</p>`),
```
Comment (pipu-odoo) [inline addons/web/static/tests/views/pivot_view.test.js:3624]: The string in the function is a constant. What's more, the function is used in a unit test. The risk is zero.

File: addons/web/static/tests/views/pivot_view.test.js [hunk 3]
```diff
@@ -0,0 +1,4551 @@
...
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        context: { search_default_small_than_0: true },
+        noContentHelp: markup('<p class="abc">click to add a foo</p>'),
```
Comment (pipu-odoo) [inline addons/web/static/tests/views/pivot_view.test.js:3650]: @odoo/rd-security The string in the function is a constant. What's more, the function is used in a unit test. The risk is zero.

File: addons/web/static/tests/views/pivot_view.test.js [hunk 4]
```diff
@@ -0,0 +1,4551 @@
...
+		<field name="date" interval="month" type="col"/>
+	</pivot>`;
+    Partner._views["search,false"] = `<search>
+		<filter name="small_than_0" string="Small Than 0" domain="[('id', '&lt;', 0)]"/>
+	</search>`;
+
+    await mountView({
+        type: "pivot",
+        resModel: "partner",
+        noContentHelp: markup('<p class="abc">click to add a foo</p>'),
```
Comment (pipu-odoo) [inline addons/web/static/tests/views/pivot_view.test.js:3676]: @odoo/rd-security The string in the function is a constant. What's more, the function is used in a unit test. The risk is zero.

**Pull request closed without merging**